lucene/analysis/common/src/test/org/apache/lucene/analysis/hunspell/TestAllDictionaries.java

```diff
@@ -160,7 +160,8 @@ public void testDictionariesLoadSuccessfully() throws Exception {
           try {
             Dictionary dic = loadDictionary(aff);
             totalMemory.addAndGet(RamUsageTester.sizeOf(dic));
-            totalWords.addAndGet(RamUsageTester.sizeOf(dic.words));
+            totalWords.addAndGet(
+                RamUsageTester.sizeOf(dic.words) + RamUsageTester.sizeOf(dic.wordHashes));
```

Comment:
About ~7% memory usage increase on average, at most 512KB

Reply:
These are actual numbers from the dictionaries you checked, correct? The code doesn't seem to have an upper limit (only highestOneBit(words*10)). I hope the dictionaries out there are reasonable (lexicon of spoken human languages certainly is bound, so I think it's a sane assumption).

Reply:
Yes, that's from wooorm and LibreOffice dictionaries